internal/namespaces/k8s/v1beta4/custom_cluster.go

```diff
@@ -40,3 +49,41 @@ func clusterAvailableVersionsListBuilder(c *core.Command) *core.Command {
 
 	return c
 }
+
+func clusterCreateBuilder(c *core.Command) *core.Command {
+	c.WaitFunc = waitForClusterFunc()
+	return c
+}
+
+func clusterDeleteBuilder(c *core.Command) *core.Command {
+	c.WaitFunc = waitForClusterFunc()
+	return c
+}
+
+func clusterUpgradeBuilder(c *core.Command) *core.Command {
+	c.WaitFunc = waitForClusterFunc()
+	return c
+}
+
+func clusterUpdateBuilder(c *core.Command) *core.Command {
+	c.WaitFunc = waitForClusterFunc()
+	return c
+}
+
+func waitForClusterFunc() core.WaitFunc {
+	return func(ctx context.Context, _, respI interface{}) (interface{}, error) {
+		cluster, err := k8s.NewAPI(core.ExtractClient(ctx)).WaitForCluster(&k8s.WaitForClusterRequest{
+			Region:    respI.(*k8s.Cluster).Region,
+			ClusterID: respI.(*k8s.Cluster).ID,
+			Timeout:   scw.DurationPtr(clusterActionTimeout),
+		})
+		if err != nil {
+			notFoundError := &scw.ResourceNotFoundError{}
+			responseError := &scw.ResponseError{}
+			if xerrors.As(err, &responseError) && responseError.StatusCode == http.StatusNotFound || xerrors.As(err, &notFoundError) {
```

Comment:
do we really need a new dependency ? 

Would it be hard to do it without `xerrors` ?

Reply:
:man_shrugging: I guess not, i'll take a look